packages/core/src/model.types.ts

```diff
@@ -37,10 +34,24 @@ export interface Model<
   events: Prop<TModelCreators, 'events'>;
   actions: Prop<TModelCreators, 'actions'>;
   reset: () => AssignAction<TContext, any>;
-  createMachine: (
-    config: MachineConfig<TContext, any, TEvent, TAction>,
-    implementations?: Partial<MachineOptions<TContext, TEvent, TAction>>
-  ) => StateMachine<TContext, any, TEvent>;
+  createMachine: {
+    <TTypesMeta extends TypegenConstraint = TypegenDisabled>(
+      config: MachineConfig<TContext, any, TEvent, TAction, TTypesMeta>,
+      implementations?: MaybeTypegenMachineOptions<
+        TContext,
+        TEvent,
+        TAction,
+        ResolveTypegenMeta<TTypesMeta, TEvent, TAction>
```

Comment:
calling `ResolveTypegenMeta` once here was impossible because of the TS issue with the inference that I've been debugging, inlining it into both call sites fixed the issue